src/main/kotlin/com/cjbooms/fabrikt/generators/controller/MicronautControllerInterfaceGenerator.kt

```diff
@@ -149,9 +166,40 @@ class MicronautControllerInterfaceGenerator(
             )
         }
 
+
+        var securityOption = op.getSecurityRequirements().securitySupport(op.hasSecurityRequirements())
+
+        if(securityOption == SecuritySupport.NO_SECURITY) {
+            securityOption = globalSecurity
+        }
+
+        val securityRule = setSecurityRule(securityOption)
+
+        if(securityRule != "") {
+            this.addAnnotation(
+                AnnotationSpec
+                    .builder(MicronautImports.SECURED)
+                    .addMember(
+                        securityRule
+                    )
+                    .build()
+            )
+        }
+
         return this
     }
 
+    private fun setSecurityRule(
```

Comment:
**suggestion:** Did you consider inlining this function? Or, in other words, what's your thought process behind making this a function?

Reply:
To keep things a bit cleaner and modular. If you prefer, I will inline it.

Reply:
To me personally, the function doesn't add any clarity to the code (the variable and type names already do enough) so I wouldn't have written it, if I were working on this issue. However, it's just an opinion, so you feel free to decide yourself.

If you want to keep the function, you should rename it though, because it's not a "setter" and "SecurityRule" doesn't tell you about its purpose unless you already know what it's there for. Maybe something like `getSecuredAnnotationArguments`.